### ISSUE TITLE
Data not updated

### ISSUE THREAD

**Issue by droberge** [arcpy/update-hosted-feature-service#12]
Hello. 

I am trying to implement this script to update one of our hosted feature service. See result below:
##########################################
C:\>C:\Python27\ArcGIS10.3\python.exe C:\droberge\downloads\esri\update_hosted_fs\update.py
Starting Feature Service publish process
found Feature Service : 112fd0d17da14216876490dd6071f3d2
found Service Definition : 7ad2e2f37bb445fb8ac68c57acfe4323
Created C:\droberge\downloads\esri\update_hosted_fs\tempDir\Ames_Utilities_webHosted.sd
updated SD:   7ad2e2f37bb445fb8ac68c57acfe4323
Problem reading response:
 'success'

finished.
##########################################

Desktop version: 10.3

when looking at the feature service on AGOL, all layers are empty

Any help would be much appreciated

Thanks!

D

**Comment by cpedrezuela** [comment]
Im also experiencing the same thing. Using 10.3 desktop and updating 140k records of a hosted service. I always get Problem reading response: 'succes', finished. And the resulting hosted service is not equal to the datasource on the mxd.

**Comment by khibma** [comment]
@droberge @cpedrezuela appologies, the error message you're getting is __really__ useless.
I just tweaked the code slightly (#13). If you download the latest copy of update.py (or just update the one line changed), it'll now spit out the actual message from when you're in the publish routine.

This wont solve your problems, but it should at least provide a better clue.

**Comment by droberge** [comment]
Thanks for updating the code, now I am getting this as the problem
*****************************************************************************
Starting Feature Service publish process
found Feature Service : 8e62b90ccd6543f3885d1f2e9eae965f
found Service Definition : 7ad2e2f37bb445fb8ac68c57acfe4323
Created C:\droberge\downloads\esri\update_hosted_fs\tempDir\Ames_Utilities_webHosted.sd
updated SD:   7ad2e2f37bb445fb8ac68c57acfe4323
Problem reading response, jsonResponse: 
 {u'services': [{u'encodedServiceURL': u'http://services.arcgis.com/GM5KudYVoaALNBe2/arcgis/rest/services/Ames_Utilities_webHosted/FeatureServer', u'jobId': u'18fc7513-8b71-41f8-a65b-64adfdc65153', u'serviceurl': u'http://services.arcgis.com/GM5KudYVoaALNBe2/arcgis/rest/services/Ames_Utilities_webHosted/FeatureServer', u'type': u'Feature Service', u'serviceItemId': u'8e62b90ccd6543f3885d1f2e9eae965f', u'size': 4947879}]}

finished.
*****************************************************************************
Any idea what the problem might be?

**Comment by khibma** [comment]
Oh wow. Yeah, I think I see the problem now. The response you've got back is a jobId.
That means the Server is currently processing the request to publish. I guess I've taken for granted that I've published 'smaller' items and publish happens really quickly.

I can probably fix that, but it'll take me a day or 2 to get the code ready and tested.

Looking at the code though - my guess is that it should have actually published the item you've uploaded. The only thing is the script wouldn't have set the sharing options on the featureservice. So while the script says there is a problem, if you check the service itself after a few moments/minutes I'd guess the feature service should be there and good.

Note to self: check status by http://resources.arcgis.com/en/help/arcgis-rest-api/index.html#/Status/02r300000089000000/

**Comment by droberge** [comment]
@ khibma , yes it did load the data, I guess I can leave with that. I'll keep an eye on the post just in case you update the code. Thanks for your hard work!

D

**Comment by khibma** [comment]
@droberge If you have a spare couple minutes, could you take the *update.py* from here: https://github.com/khibma/update-hosted-feature-service/blob/master/update.py and test it in your workflow? You should now see messages about the status.
I'm 90% sure of this fix, but to hear that it works for someone else would be good.

If it works for you, I'll merge it from my github into the arcpy one.

thanks
```
Checking the status of publish..
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  completed

item finished published
```

**Comment by droberge** [comment]
you did it!!!!! Thank you so much for your hard work. This is slick

Starting Feature Service publish process
found Feature Service : 8e62b90ccd6543f3885d1f2e9eae965f
found Service Definition : 7ad2e2f37bb445fb8ac68c57acfe4323
Created C:\droberge\downloads\esri\update_hosted_fs\tempDir\Ames_Utilities_webHosted.sd
updated SD:   7ad2e2f37bb445fb8ac68c57acfe4323
Checking the status of publish..
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  processing
  completed
item finished published

finished.

**Comment by khibma** [comment]
Great, thanks for confirming. I'll merge it into this as soon as I figure out how to solve my conflicts :(

**Comment by khibma** [comment]
fixed in PR #15
